src/library/libraryview.cpp

```diff
@@ -709,6 +737,29 @@ void LibraryView::FilterReturnPressed() {
   emit doubleClicked(currentIndex());
 }
 
+void LibraryView::SearchForArtist() {
+  SaveFocus();
+  if (!last_selected_song_.albumartist().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.albumartist().simplified());
+  } else if (!last_selected_song_.artist().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.artist().simplified());
+  }
+}
+
+void LibraryView::SearchForAlbum() {
+  SaveFocus();
+  if (!last_selected_song_.album().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.album().simplified());
+  }
+}
+
+void LibraryView::SearchForThis() {
+  SaveFocus();
+  if (!last_selected_container_.isEmpty()) {
+    filter_->ShowInLibrary(last_selected_container_.simplified());
```

Comment:
It would be pretty weird if a user could initiate a search from there that resulted in zero results.